pkg/monitor/monitor.go

```diff
@@ -50,17 +49,20 @@ func Monitor(clusterName, clusterDomain, templatePath, cfgPath, apiVip string, a
 	}
 	defer conn.Close()
 
-	domain := fmt.Sprintf("%s.%s", clusterName, clusterDomain)
 	log.Info("API is not reachable through HAProxy")
 	for {
 		select {
 		case <-done:
 			cleanHAProxyPreRoutingRule(apiVip, apiPort, lbPort)
 			return nil
 		default:
-			config, err := config.GetLBConfig(domain, apiPort, lbPort, statPort, net.ParseIP(apiVip))
+			config, err := config.GetLBConfig(kubeconfigPath, apiPort, lbPort, statPort, net.ParseIP(apiVip))
 			if err != nil {
-				return err
+				log.WithFields(logrus.Fields{
+					"kubeconfigPath": kubeconfigPath,
+				}).Info("GetLBConfig failed, sleep half of interval and retry")
+				time.Sleep(interval / 2)
```

Comment:
I noticed that haproxy-moniotr container exits (and restarted by kubelet) , the root cause was  'GetLBConfig' failed to read nodes via OCP API ( error was something like 'etcd leader changed').
So, since we implemented a de-bounce threshold in the monitor code, I think we should ignore this poll instead of exiting, and also it make sense to start the next poll faster. I set it to half the default interval, but I'm totally open for other ideas.

Reply:
Yeah, I understand why a retry is needed here, I just wasn't sure why we needed a shorter interval. I'm not concerned about it, but I wanted to see if I was missing something.